src/utilities/params.ts

```diff
@@ -45,14 +44,14 @@ const numberParam: ParamGetSet = {
   },
   set: (value, { invalid }) => {
     if (typeof value !== 'number') {
-      invalid()
+      throw invalid()
     }
 
     return value.toString()
   },
 }
 
-export function getParamValue(value: string, param: Param): unknown {
+export function getParamValue<T extends Param>(value: string, param: T): unknown {
```

Comment:
Why not keep the overload with the strict type? Did the strict type cause issues where this was being used?

Reply:
by overload do you mean the generic? I assumed you would wanna drop it since it's not actually used for anything anymore with the return type being `unknown`

Reply:
My thought was the overload is the correct type and the implementation is unknown just because typescript won't narrow the generic inside the implementation. But when you call this function with a value and a param it correctly tells you what type it will return from the overload.

Reply:
In my commit this function was overloaded with a single overload that had the return type to extract the param return value. That becomes the exported signature. And the implementation uses unknown

Reply:
nice! I somehow missed that